jenkins-integration/beaker/install/shared/45_install_hiera_config.rb

```diff
@@ -0,0 +1,39 @@
+require 'puppet/gatling/config'
+require 'tmpdir'
+require 'yaml'
+
+test_name 'Install hiera config'
+
+def target_hiera_config_path(host)
+  return on(host, puppet('config print hiera_config')).stdout.chomp
+end
+
+def get_source_hiera_config(host, hiera_config)
+  source_hiera_config_path = hiera_config[:source_file]
+  if source_hiera_config_path.nil? || source_hiera_config_path.empty?
+    source_hiera_config_path = target_hiera_config_path(host)
```

Comment:
The code as written in the PR would allow a job config to specify a datadir without specifying a full-blown hiera.yaml to apply first.  In that mode, we'd basically end up using the default hiera.yaml file laid down at package install time but possibly with the `datadir` key in it set to pick up hieradata from some other place.  The code above just defaults to pulling down the package-installed hiera.yaml file for the case that a job configuration would not provide its own.

I understand this is probably a very unlikely case that we'd need to support since we're usually going to have a custom hierarchy / custom hieradata or no hiera at all for whatever code configurations we're going to use.  If it bugs you too much, I suppose I could get rid the defaulting logic.  Not a big deal to me either way.